cmd/rds-init/main.go

```diff
@@ -0,0 +1,299 @@
+		dbname = user
+	}
+
+	return &dbConnInfo{
+		Host:   host,
+		Port:   port,
+		DBName: dbname,
+		User:   user,
+		RawDSN: dsn,
+	}, nil
+}
+
+func loadAWSConfig(ctx context.Context) (aws.Config, error) {
+	return awsconfig.LoadDefaultConfig(ctx)
+}
+
+func getSecretIfExists(ctx context.Context, sm *secretsmanager.Client, name string) (*nessusSecret, error) {
+	out, err := sm.GetSecretValue(ctx, &secretsmanager.GetSecretValueInput{
+		SecretId: aws.String(name),
+	})
+	if err != nil {
+		var rnfe *smtypes.ResourceNotFoundException
+		if errors.As(err, &rnfe) {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("GetSecretValue failed: %w", err)
+	}
+
+	if out.SecretString == nil {
+		return nil, fmt.Errorf("existing secret %q does not have SecretString", name)
+	}
+
+	var sec nessusSecret
+	if err := json.Unmarshal([]byte(*out.SecretString), &sec); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal existing secret JSON: %w", err)
+	}
+	return &sec, nil
+}
+
+func createSecret(ctx context.Context, sm *secretsmanager.Client, name string, sec *nessusSecret) error {
+	payload, err := json.Marshal(sec)
+	if err != nil {
+		return fmt.Errorf("failed to marshal secret JSON: %w", err)
+	}
+	_, err = sm.CreateSecret(ctx, &secretsmanager.CreateSecretInput{
+		Name:         aws.String(name),
+		SecretString: aws.String(string(payload)),
+	})
+	if err != nil {
+		return fmt.Errorf("CreateSecret failed: %w", err)
+	}
+	return nil
+}
+
+func findDBInstanceByEndpoint(ctx context.Context, client *rds.Client, host string, port int32) (string, error) {
+	var marker *string
+
+	for {
+		out, err := client.DescribeDBInstances(ctx, &rds.DescribeDBInstancesInput{
+			Marker: marker,
+		})
+		if err != nil {
+			return "", fmt.Errorf("DescribeDBInstances failed: %w", err)
+		}
+
+		for _, inst := range out.DBInstances {
+			if inst.Endpoint == nil {
+				continue
+			}
+			if aws.ToString(inst.Endpoint.Address) == host && aws.ToInt32(inst.Endpoint.Port) == port {
+				return aws.ToString(inst.DBInstanceIdentifier), nil
+			}
+		}
+
+		if out.Marker == nil || len(out.DBInstances) == 0 {
+			break
+		}
+		marker = out.Marker
+	}
+
+	return "", fmt.Errorf("no RDS DB instance found with endpoint %s:%d", host, port)
+}
+
+func main() {
+	log.SetFlags(0)
+
+	if len(os.Args) != 2 {
+		log.Fatalf("Usage: %s <postgres-dsn>", os.Args[0])
+	}
+	dsn := os.Args[1]
+
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+	defer cancel()
+
+	connInfo, err := parseDSN(dsn)
+	if err != nil {
+		log.Fatalf("Error parsing DSN: %v", err)
+	}
+
+	db, err := sqlx.Connect("postgres", dsn)
+	if err != nil {
+		log.Fatalf("Error connecting to database: %s", err)
+	}
+	defer db.Close()
+
+	awsCfg, err := loadAWSConfig(ctx)
+	if err != nil {
+		log.Fatalf("Failed to load AWS config: %v", err)
+	}
+
+	smClient := secretsmanager.NewFromConfig(awsCfg)
+	rdsClient := rds.NewFromConfig(awsCfg)
+
+	portInt := 5432
+	fmt.Sscanf(connInfo.Port, "%d", &portInt)
```

Comment:
can you make the `Port` field an `int` instead, and use `strconv.Atoi` (and check the error) in `parseDSN`?